core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -979,6 +979,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     // A Hadoop configuration can be about 10 KB, which is pretty big, so broadcast it.
     val confBroadcast = broadcast(new SerializableConfiguration(hadoopConfiguration))
     val setInputPathsFunc = (jobConf: JobConf) => FileInputFormat.setInputPaths(jobConf, path)
+    clean(setInputPathsFunc)
```

Comment:
@rajeshbalamohan catching up here: I think the remaining TODO is that cleaning still needs to be restored for `hadoopRDD` right? then this is ready.